### DISCUSSION TITLE
Database url comparison

### DISCUSSION THREAD

**Discussion by IvanIsak2000** [fastapi/fastapi#9476] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#server.py
from fastapi import FastAPI, Body, status
from fastapi.responses import HTMLResponse

app = FastAPI()

fake_data = ['Iwan123]

??????????????????????
```

### Description

Ok, I have a fake  data
Let's say a user wants to log into an account - for this he goes into a special script and writes something like this:
`>> python client.py <username> <usertoken>`

After that, the script sends data to the server with the user's name and token, and the server must compare whether it converges if the name and token are added, if yes,
`return HTMLResponse('OK')`

![image](https://user-images.githubusercontent.com/79650307/235643989-ff1fce02-4037-4f35-bd7b-b319ea9da5db.png)

![image](https://user-images.githubusercontent.com/79650307/235645941-65fe0dea-cd51-4c62-9bcb-51d6e8cc7aeb.png)

I enter into the terminal, where the first word is `usernam` and the second word is `usertoken`, the script adds up these words, i.e. `Iwan123` and sends a request to the server

![image](https://user-images.githubusercontent.com/79650307/235644366-60aed762-2151-4530-91f3-e00cebec84e0.png)

<hr>

And how can I make `server.py` have one construction `@app.get.... ` and that it accepts all possible names + tokens, and depending on if they are in `fake_data` send `return HTMLResponse ('OK')`

P.S. 
I tried different ways to do it, the first and obvious solution would be:
`@app.get("/log_in/Iwan123")....`

This construction works, but if there are a thousand users, then the construction needs to look for the name + token in `fake_data`

### Operating System

Linux, Windows

### FastAPI Version

0.95.1

### Python Version

3.11.3

**Comment by 9en9i** [comment]
Hello, I think this is what you are looking for:

https://fastapi.tiangolo.com/tutorial/path-params/

**Reply by 9en9i** [reply, marked as answer]
Also in the official documentation there is a tutorial on how to do authorization: 
https://fastapi.tiangolo.com/tutorial/security. 
It also has a more advanced tutorial: 
https://fastapi.tiangolo.com/advanced/security/

**Reply by IvanIsak2000** [reply]
> Hello, I think this is what you are looking for:
> 
> https://fastapi.tiangolo.com/tutorial/path-params/

Thanks for your reply!
I'm just a beginner, I looked at how to correctly authorize a user in fast api, but for me it's still difficult

During testing, I wanted to approve authorization if the request code == 200, I spent the whole day thinking how to solve this, because even with incorrect data, the account is successful. I noticed,
that if the check inside the fast api function is skipped - returns None, then the fast api returns this as True and therefore approves the authorization.

I came up with a check from `response.context`:
```
        if fake_data[username]['token'] == usertoken:
                return True
        else:
                return False

```
To what extent is this approach inappropriate / appropriate and why?

**Reply by 9en9i** [reply]
@IvanIsak2000 I don't really understand where in your code `response.context`. I recommend that you use at least the simplest authorization method from the tutorials. In any case, you need to raise HTTPException with the correct http code (401). Example of such code: https://fastapi.tiangolo.com/tutorial/handling-errors/

**Reply by IvanIsak2000** [reply]
Thanks for the recommendation! I improved the check system: now we need to get the status 200 from the server and `response.content` should contain the bit string `b'true`

![image](https://user-images.githubusercontent.com/79650307/236115607-02fa6e63-36de-4a2c-bbf2-e2a611c982f6.png)
![image](https://user-images.githubusercontent.com/79650307/236115641-8037c416-3b6d-4b47-8e12-8b4c4a35e49f.png)

**Reply by nymous** [reply]
Hi @IvanIsak2000!
Another recommendation I would make would be to not use a GET request for authentication, as the request path will certainly be logged by the server and any proxy in between, which would leak all usernames and tokens in the logs.
You should use a POST request, and pass the username and token in the body (as JSON or form params, as you wish).

**Reply by IvanIsak2000** [reply]
> Hi @IvanIsak2000! Another recommendation I would make would be to not use a GET request for authentication, as the request path will certainly be logged by the server and any proxy in between, which would leak all usernames and tokens in the logs. You should use a POST request, and pass the username and token in the body (as JSON or form params, as you wish).

Much for your recommendation! I will defiantly listen to your advice!
